tools/linter/adapters/newlines_linter.py

```diff
@@ -33,78 +34,86 @@ class LintMessage(NamedTuple):
+                )
+
             return LintMessage(
                 path=filename,
                 line=None,
                 char=None,
                 code=LINTER_CODE,
                 severity=LintSeverity.ERROR,
-                name="testestTrailing newline",
-                original=None,
-                replacement=None,
+                name="Trailing newline",
+                original=original,
+                replacement=original.rstrip("\n") + "\n",
                 description="Trailing newline found. Run `lintrunner --take NEWLINE -a` to apply changes.",
             )
-
-        else:
-            # Read the last two bytes
-            f.seek(-2, os.SEEK_END)
-            b, c = f.read(2)
-            # no ASCII byte is part of any non-ASCII character in UTF-8
-            if b != NEWLINE and c == NEWLINE:
-                return None
-            else:
-                f.seek(0)
-                try:
-                    original = f.read().decode("utf-8")
-                except Exception as err:
-                    return LintMessage(
-                        path=filename,
-                        line=None,
-                        char=None,
-                        code=LINTER_CODE,
-                        severity=LintSeverity.ERROR,
-                        name="Decoding failure",
-                        original=None,
-                        replacement=None,
-                        description=f"utf-8 decoding failed due to {err.__class__.__name__}:\n{err}",
-                    )
-
+    for idx, line in enumerate(lines):
+        if len(line) >= 2 and line[-1] == NEWLINE and line[-2] == CARRIAGE_RETURN:
+            try:
+                decoded_line = line.decode("utf-8")
+            except Exception as err:
                 return LintMessage(
                     path=filename,
-                    line=None,
+                    line=idx,
                     char=None,
                     code=LINTER_CODE,
                     severity=LintSeverity.ERROR,
-                    name="Trailing newline",
-                    original=original,
-                    replacement=original.rstrip("\n") + "\n",
-                    description="Trailing newline found. Run `lintrunner --take NEWLINE -a` to apply changes.",
+                    name="Decoding failure",
+                    original=None,
+                    replacement=None,
+                    description=f"utf-8 decoding failed due to {err.__class__.__name__}:\n{err}",
                 )
+            return LintMessage(
+                path=filename,
+                line=idx,
+                char=None,
+                code=LINTER_CODE,
+                severity=LintSeverity.ERROR,
+                name="DOS newline",
+                original=decoded_line,
+                replacement=decoded_line[:-2] + "\n",
+                description="DOS newline found. Run `lintrunner --take NEWLINE -a` to apply changes.",
```

Comment:
```suggestion
                description="DOS newline found. Run `lintrunner --take NEWLINE -a` to standardize newlines format.",
```

Reply:
Hmm, let's not make changes to description that is used throughout the file (if you want to make it, please do in a followup PR)